linkerd/app/inbound/src/http/router.rs

```diff
@@ -246,47 +246,32 @@ impl<C> Inbound<C> {
 
 // === impl LogicalPerRequest ===
 
-impl<T> From<(policy::ServerPermit, T)> for LogicalPerRequest
+impl<T> From<(policy::RoutePermit, T)> for LogicalPerRequest
 where
     T: Param<Remote<ServerAddr>>,
     T: Param<tls::ConditionalServerTls>,
 {
-    fn from((permit, t): (policy::ServerPermit, T)) -> Self {
-        let labels = vec![
-            (
-                "srv_group".to_string(),
-                permit.labels.server.0.group().to_string(),
-            ),
-            (
-                "srv_kind".to_string(),
-                permit.labels.server.0.kind().to_string(),
-            ),
-            (
-                "srv_name".to_string(),
-                permit.labels.server.0.name().to_string(),
-            ),
-            (
-                "authz_group".to_string(),
-                permit.labels.authz.group().to_string(),
-            ),
-            (
-                "authz_kind".to_string(),
-                permit.labels.authz.kind().to_string(),
-            ),
-            (
-                "authz_name".to_string(),
-                permit.labels.authz.name().to_string(),
-            ),
-        ];
+    fn from((permit, t): (policy::RoutePermit, T)) -> Self {
+        let labels = [
+            ("srv", &permit.labels.route.server.0),
+            ("route", &permit.labels.route.route),
+            ("authz", &permit.labels.authz),
+        ]
+        .into_iter()
+        .flat_map(|(k, v)| {
+            [
+                (format!("{k}_group"), v.group().to_string()),
+                (format!("{k}_kind"), v.kind().to_string()),
+                (format!("{k}_name"), v.name().to_string()),
```

Comment:
i was kind of bummed out that we had to `to_string`-ify all the meta bits here where they could be `Arc`ed and just implement `FormatLabels` everywhere else...but i realized this is because of the current way `tap::Labels` works. might be worth messing with that separately i guess...